gc/src/trace.rs

```diff
@@ -1,5 +1,18 @@
+/// The Finalize trait. Can be specialized for a specific type to define
+/// finalization logic for that type.
+pub trait Finalize {
+    fn finalize(&self);
```

Comment:
Is there a reason to have this method in its own trait instead of making it a defaulted method of `Trace`?